### PR TITLE
Migrate ci

### PR DESCRIPTION
**Pull request by darrencl**
With its current version, Travis CI is not free anymore, so right now our CI could not run.

This PR will migrate suspect's CI to GitHub actions, which includes:

* `ci-workflows.yaml` to run tests
* `publish-pypi.yaml` to publish to main PyPI. This is mainly following [Python Packaging User Guide](https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/).

### CODE REVIEW

File: .github/workflows/publish-pypi.yaml
Comment (darrencl) [inline .github/workflows/publish-pypi.yaml:19]: @bennyrowland The build job is using the newest python 3 version. Let me know if you think it is best to pin the version to 3.8 (same version as test job in `ci-workflows.yaml`)

Comment (bennyrowland) [reply .github/workflows/publish-pypi.yaml:19]: I think this is probably fine, if it passes the tests under 3.8 then it doesn't really matter what is used to build it. And we probably don't need to continue to support 3.8 if we don't want to.

File: .github/workflows/ci-workflows.yaml
Comment (bennyrowland) [inline .github/workflows/ci-workflows.yaml:28]: What is `requirements_dev.txt`?

File: .github/workflows/publish-pypi.yaml
Comment (bennyrowland) [inline .github/workflows/publish-pypi.yaml:44]: Does this need an added secret to make this work? I haven't done much with GHA.

**Review by bennyrowland — APPROVED** [review]
I don't know much about GHA but this looks basically fine to me - I have added a couple of questions but am happy for this to progress. It may well be that there are a few cycles of fiddling with the workflows to actually get everything working, sometimes you just have to try it and see what errors come up. Is there anything practical that you need from me to complete this, or just my blessing (which you have)?

File: .github/workflows/ci-workflows.yaml
Comment (darrencl) [reply .github/workflows/ci-workflows.yaml:28]: Good catch! Seems suspect uses only `requirements.txt` — I will fix this.

Some package maintains a separate `requirements_dev.txt`, which contains more dependencies that are only relevant to the development environment.

File: .github/workflows/publish-pypi.yaml
Comment (darrencl) [reply .github/workflows/publish-pypi.yaml:44]: This is the first time for me as well on setting up the PyPI publishing from GHA.

From reading the link below, it uses Open ID Connect which we need to set up in PyPI's trusted publishing settings page.

https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-pypi#adding-the-identity-provider-to-pypi

**Comment by darrencl** [comment]
@bennyrowland Thank you for reviewing! I will fix that `requirement.txt` and setup PyPI's trusted publisher to make this work. 

Nothing practical, I just need another pair of eyes to check as I do not want to pollute the repository you worked so hard on. :smile:

**Pull request merged**
